### ISSUE TITLE
Async broker initialization

### ISSUE THREAD

**Issue by austinkelleher** [SocketCluster/socketcluster#351]
I tried to sift through the documentation and source code, but I couldn't seem to find a way to initialize a broker asynchronously. For example, I have an asynchronous task that I'd like to execute in my broker's `run` function. I'm new to socketcluster, so perhaps this is an anti-pattern.

My specific use case is I'd like for my Redis cluster clients to explicitly report that they're ready. Something like this:

```js
class Broker extends SCBroker {
  async run () {
    await scRedis.attach(this)
    // All Redis cluster clients are ready!
    ...
  }
}
```   

Thoughts @jondubois?

Thanks in advance.

**Comment by jondubois** [comment]
@austinkelleher This pattern is not currently supported but it should be easy to add support for it. Please leave this issue open.

In the meantime, an alternative to this is to delay instantiation of the broker.
E.g. You can move this line https://github.com/SocketCluster/socketcluster/blob/25b3f84f2bb69f92f9f9c404167c23ea27e5e919/sample/broker.js#L27 in a new async function and do all your `await` operations before you instantiate the broker.

**Comment by jondubois** [comment]
@austinkelleher Ok, this feature should now be available in SC v9.3.0 https://github.com/SocketCluster/socketcluster/releases/tag/v9.3.0

**Comment by austinkelleher** [comment]
@jondubois Thanks a lot! I really appreciate you getting the fix in so quickly. I tried it out and everything works great except for error reporting. If either the broker's `run` method or starting the HTTP server reject with an error, it is not getting propagated properly. I opened up these two PRs to fix the issues:

https://github.com/SocketCluster/socketcluster/pull/353
https://github.com/SocketCluster/sc-broker/pull/5
